python/feathub/processors/local/local_processor.py

```diff
@@ -178,26 +210,55 @@ def materialize_features(
                 timestamp_field=features.timestamp_field,
                 timestamp_format=features.timestamp_format,
             )
-        elif isinstance(sink, FileSystemSink):
+        elif isinstance(sink, FileSystemSink) and _is_local_file_or_dir(sink.path):
             insert_into_file_sink(features_df, sink)
             return LocalJob()
+        elif is_spark_supported_sink(type(sink)):
```

Comment:
Would it be simpler to invoke this method as `is_spark_supported_sink(sink)`?

Same for `is_spark_supported_source(...)`.